### PR TITLE
Level-Up form Bugfixes

### PR DESCRIPTION
**Pull request by Ellamental2**
fixed a bug where level-up-points sometimes show as undefined if baseStatModifiers is undefined for any reason.

in the case that the modifier is undefined the system will now assume it to be 0, updating the actor should then (in theory) define the modifiers

I have also edited the way the missing health is calculated and applied so when a pokemon's maximum health goes up the amount of health they are missing stays the same as opposed to the current health staying the same. Current health will go up the same amount as max health.

Previous health calculation was causing an error in the _updateObject() function and causing it to not work when health of the mon changed.

### CODE REVIEW

**Review by dylanpiera — CHANGES_REQUESTED** [review]
need to check changes in level-up-form

File: module/forms/level-up-form.js [hunk 1]
```diff
@@ -102,7 +102,18 @@ export class PTULevelUpForm extends FormApplication {
...
-    const lostHealth = this.object.actor.system.health.max - this.object.actor.system.health.value;
+    const oldData = {
+      system: {
+        level:{
+          "current": state.changeDetails.oldLvl,
+          "exp": state.changeDetails.oldExp
+        }
+      }
+    }
+    await this.object.actor.update(oldData);
```
Comment (dylanpiera) [inline module/forms/level-up-form.js:113]: Why are you updating the actor here to something it already is?

Also `current` should never be directly set, it's a derrived value.

Comment (dylanpiera) [reply module/forms/level-up-form.js:113]: I just read your comment below; yeah let's not do this, way too many database calls for a very simple function that realistically doesn't *do* much in the grand scheme of this.

Either manually calculate the new health value, or don't bother

File: module/forms/level-up-form.js [hunk 2]
```diff
@@ -124,7 +136,10 @@ export class PTULevelUpForm extends FormApplication {
           "spdef.levelUp": (state.evolving.is ? 0 : state.stats.spdef.levelUp) + (state.stats.spdef.newLevelUp ?? 0),
           "spd.levelUp": (state.evolving.is ? 0 : state.stats.spd.levelUp) + (state.stats.spd.newLevelUp ?? 0),
         },
-        // 'health.value': this.object.actor.system.health.max - lostHealth
+        level : {
+          "current": state.changeDetails.newLvl,
```
Comment (dylanpiera) [inline module/forms/level-up-form.js:140]: Don't update current.

**Review by dylanpiera — CHANGES_REQUESTED** [review]

File: module/forms/level-up-form.js (hunk 2 above)
Comment (Ellamental2) [reply module/forms/level-up-form.js]: new commit removes this

File: module/forms/level-up-form.js (hunk 1 above)
Comment (Ellamental2) [reply module/forms/level-up-form.js]: manually calculated old health values to get missing health, then manually calculated the new max health then subtracted the missing health to get the new current health

```js
const oldHealthTotal = 10 + state.changeDetails.oldLvl + (this.object.actor.system.stats.hp.total * 3);
const oldHealthMax = this.object.actor.system.health.injuries > 0 ? Math.trunc(oldHealthTotal * (1 - ((this.object.actor.system.modifiers.hardened ? Math.min(this.object.actor.system.health.injuries, 5) : this.object.actor.system.health.injuries) / 10))) : oldHealthTotal;
const oldHealthValue = this.object.actor.system.health.value ?? oldHealthMax;

const missinghealth = oldHealthMax - oldHealthValue;
```
   


```js
//adjust health as per level up
const newHealthTotal = 10 + state.changeDetails.newLvl + (this.object.actor.system.stats.hp.total * 3);
const newHealthMax = this.object.actor.system.health.injuries > 0 ? Math.trunc(newHealthTotal * (1 - ((this.object.actor.system.modifiers.hardened ? Math.min(this.object.actor.system.health.injuries, 5) : this.object.actor.system.health.injuries) / 10))) : newHealthTotal;
const newhealth = newHealthMax - missinghealth;
```

**Review by dylanpiera — APPROVED** [review]

**Pull request merged**